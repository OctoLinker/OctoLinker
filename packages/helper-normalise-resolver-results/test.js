import normaliseResolverResults from './index.js';

describe('normaliseResolverResults', () => {
  test.each([
    '{BASE_URL}file.js',
    '{BASE_URL}/file.js',
    '{BASE_URL}/foo/bar/blob/master/file.js',
    '{BASE_URL}/foo/bar/blob/1ab8cfd3b65d3b43335130d6cefbf8c62482680f/file.js',
    'https://foosearch.org/',
    'https://novalidurl',
    ['{BASE_URL}/foo/bar/blob/master/file.js'],
    [
      '{BASE_URL}/foo/bar/blob/master/file.js',
      '{BASE_URL}/foo/bar/blob/master/file.js',
    ],
    { registry: 'npm', target: 'foo' },
    () => {},
  ])('converts %s', (url) => {
    expect(normaliseResolverResults([].concat(url))).toMatchSnapshot();
  });
});
