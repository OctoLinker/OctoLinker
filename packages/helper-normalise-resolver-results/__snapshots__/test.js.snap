// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normaliseResolverResults converts [object Object] 1`] = `
Array [
  Object {
    "registry": "npm",
    "target": "foo",
    "type": "registry",
  },
]
`;

exports[`normaliseResolverResults converts {BASE_URL}/file.js 1`] = `
Array [
  Object {
    "branch": undefined,
    "path": undefined,
    "repo": undefined,
    "type": "internal-link",
    "url": "https://github.com/file.js",
    "user": "file.js",
  },
]
`;

exports[`normaliseResolverResults converts {BASE_URL}/foo/bar/blob/1ab8cfd3b65d3b43335130d6cefbf8c62482680f/file.js 1`] = `
Array [
  Object {
    "branch": "1ab8cfd3b65d3b43335130d6cefbf8c62482680f",
    "path": "file.js",
    "repo": "bar",
    "type": "internal-link",
    "url": "https://github.com/foo/bar/blob/1ab8cfd3b65d3b43335130d6cefbf8c62482680f/file.js",
    "user": "foo",
  },
]
`;

exports[`normaliseResolverResults converts {BASE_URL}/foo/bar/blob/master/file.js 1`] = `
Array [
  Object {
    "branch": "master",
    "path": "file.js",
    "repo": "bar",
    "type": "internal-link",
    "url": "https://github.com/foo/bar/blob/master/file.js",
    "user": "foo",
  },
]
`;

exports[`normaliseResolverResults converts {BASE_URL}file.js 1`] = `
Array [
  Object {
    "branch": undefined,
    "path": undefined,
    "repo": undefined,
    "type": "internal-link",
    "url": "https://github.comfile.js",
    "user": undefined,
  },
]
`;

exports[`normaliseResolverResults converts function () {} 1`] = `
Array [
  Object {
    "handler": [Function],
    "type": "function",
  },
]
`;

exports[`normaliseResolverResults converts https://foosearch.org/ 1`] = `
Array [
  Object {
    "target": "https://foosearch.org/",
    "type": "ping",
  },
]
`;
